pages/api/sequences.py

```diff
@@ -235,64 +234,49 @@ def add_element(self, request, slug=None):
+                return api_response.Response(
+                    {'detail': 'element added'}, status=status.HTTP_201_CREATED)
             except IntegrityError:
-                return api_response.Response({'detail':
-                  'An element already exists on this rank for this sequence.'},
-                  status=status.HTTP_400_BAD_REQUEST)
-        return api_response.Response({'detail': 'invalid parameters'},
-                                     status=status.HTTP_400_BAD_REQUEST)
-
-    @action(detail=True, methods=['post'])
-    def remove_element(self, request, ):
+                return api_response.Response(
+                    {'detail':
+                         'An element already exists at this rank for this sequence.'},
+                          status=status.HTTP_400_BAD_REQUEST)
+        return api_response.Response(
+            {'detail': 'invalid parameters'}, status=status.HTTP_400_BAD_REQUEST)
+
+    @action(detail=True, methods=['delete'], url_path='elements/(?P<element_rank>\\d+)')
+    def remove_element(self, request, slug=None, element_rank=None):
         """
         - **Remove Element from Sequence**
 
-        Removes an element from a sequence. Requires a specified rank and page_element
-        to delete the EnumeratedElements instance.
+        Removes an element from a sequence by its rank.
 
         .. code-block:: http
 
-            POST /api/sequences/1/remove_element HTTP/1.1
+            DELETE /api/sequences/sequence1/elements/1 HTTP/1.1
             Content-Type: application/json
 
-            {
-                "page_element": production,
-                "rank": 1
-            }
-
         responds
 
         .. code-block:: json
 
             {
                 "detail": "element removed"
             }
-
         """
-        #pylint:disable=unused-argument
         sequence = self.get_object()
-        serializer = self.get_serializer(data=request.data)
-        if serializer.is_valid():
-            deleted, _ = EnumeratedElements.objects.filter(
-                sequence=sequence,
-                page_element=serializer.validated_data['page_element'],
-                rank=serializer.validated_data['rank']
-            ).first().delete()
-
-            if deleted:
-                return api_response.Response({'detail': 'element removed'},
-                                             status=status.HTTP_200_OK)
-            return api_response.Response({'detail': 'element at the specified rank '
-                                                    'not found in sequence'},
-                                         status=status.HTTP_404_NOT_FOUND)
-        return api_response.Response({'detail': 'invalid parameters'},
-                                     status=status.HTTP_400_BAD_REQUEST)
+        if element_rank is not None:
+            try:
+                element = EnumeratedElements.objects.get(sequence=sequence, rank=element_rank)
+                element.delete()
```

Comment:
Note to self: will have to test what happens when browsing through a sequence where ranks have gaps.